modules/stitching/src/matchers.cpp

```diff
@@ -583,7 +583,10 @@ void SurfFeaturesFinderGpu::find(InputArray image, ImageFeatures &features)
     image_.upload(image);
 
     ensureSizeIsEnough(image.size(), CV_8UC1, gray_image_);
-    cvtColor(image_, gray_image_, COLOR_BGR2GRAY);
+
+    Mat gray_image;
+    cvtColor(image, gray_image, COLOR_BGR2GRAY);
+    gray_image_.upload(gray_image);
```

Comment:
HI,

In fact the line number 583 has no impact on the code and wasting additional transfer time. 
LN 583:` image_.upload(image);`

Since, the surf has nothing to do with the colored version of the image, we are sending it to GPU unnecessarily. So, I would recommend removing this line as well. In the end, we are compromising the execution time without any benefits.